keyboards/satan/keymaps/default/keymap.c

```diff
@@ -64,25 +66,19 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
   #endif
 };
 
-enum function_id {
-    SHIFT_ESC,
-};
+bool process_record_user(uint16_t keycode, keyrecord_t* record) {
+  static uint8_t shift_esc_shift_mask;
 
-const uint16_t PROGMEM fn_actions[] = {
-  [0]  = ACTION_FUNCTION(SHIFT_ESC),
-};
+  switch (keycode) {
+    case SFT_ESC:
+      shift_esc_shift_mask = get_mods() & MODS_CTRL_MASK;
 
-void action_function(keyrecord_t *record, uint8_t id, uint8_t opt) {
-  static uint8_t shift_esc_shift_mask;
-  switch (id) {
-    case SHIFT_ESC:
-      shift_esc_shift_mask = get_mods()&MODS_CTRL_MASK;
       if (record->event.pressed) {
         if (shift_esc_shift_mask) {
           add_key(KC_GRV);
           send_keyboard_report();
         } else {
-          add_key(KC_ESC);
+          add_key(KC_GRV);
```

Comment:
You sure about this?

Reply:
Come to think of it, it looks like this entire `SFT_ESC` keycode can be replaced with `KC_GESC` with a couple of overrides defined (`GRAVE_ESC_CTRL_OVERRIDE`, `GRAVE_ESC_GUI_OVERRIDE`, `GRAVE_ESC_ALT_OVERRIDE`)

Reply:
Whoops. I rewrote some of these by hand before realising that was a dumb idea and just changed the function name and signature.

I'm trying not to change stuff too much because after all, these keymaps aren't mine ;)